File: src/main/java/com/vspk/bookrest/service/UserService.java
package com.vspk.bookrest.service;

import com.vspk.bookrest.domain.User;

import java.util.List;
import java.util.Optional;

public interface UserService {
    User save(User user);

    List<User> getAll();

    Optional<User> findByUsername(String username);

    Optional<User> findByEmail(String email);

    Optional<User> findById(Long id);

    void incrementVerificationTimesAsked(Long userId);

    void delete(Long id);

    //for tests purposes only
    void deleteAll();
}

File: src/main/java/com/vspk/bookrest/service/UserService.kt
package com.vspk.bookrest.service

import com.vspk.bookrest.domain.User
import java.util.*

interface UserService {
    fun save(user: User): User
    fun getAll(): List<User>
    fun findByUsername(username: String): Optional<User>
    fun findByEmail(email: String): Optional<User>
    fun findById(id: Long): Optional<User>
    fun incrementVerificationTimesAsked(userId: Long)
    fun delete(id: Long)

    //for tests purposes only
    fun deleteAll()
}